stellargraph/mapper/sequences.py

```diff
@@ -653,25 +653,48 @@ class CorruptedNodeSequence(Sequence):
 
     def __init__(self, base_generator):
 
-        if not isinstance(base_generator, (FullBatchSequence, SparseFullBatchSequence)):
+        self.base_generator = base_generator
+
+        if isinstance(base_generator, (FullBatchSequence, SparseFullBatchSequence)):
+            self.targets = np.zeros((1, len(base_generator.target_indices), 2))
+            self.targets[0, :, 0] = 1.0
+        elif isinstance(base_generator, NodeSequence):
+            self.targets = np.zeros((base_generator.batch_size, 2))
+            self.targets[:, 0] = 1.0
+        else:
             raise TypeError(
-                f"base_generator: expected FullBatchSequence or SparseFullBatchSequence, "
-                f"found {type(base_generator).__name__}"
+                f"base_generator: expected FullBatchSequence, SparseFullBatchSequence, "
+                f"or NodeSequence, found {type(base_generator).__name__}"
             )
 
-        self.base_generator = base_generator
-        self.targets = np.zeros((1, len(base_generator.target_indices), 2))
-        self.targets[0, :, 0] = 1.0
-
     def __len__(self):
+
         return len(self.base_generator)
 
     def __getitem__(self, index):
 
         inputs, _ = self.base_generator[index]
-        features = inputs[0]
 
-        shuffled_idxs = np.random.permutation(features.shape[1])
-        shuffled_feats = features[:, shuffled_idxs, :]
+        if isinstance(
+            self.base_generator, (FullBatchSequence, SparseFullBatchSequence)
+        ):
+
+            features = inputs[0]
+            shuffled_idxs = np.random.permutation(features.shape[1])
+            shuffled_feats = features[:, shuffled_idxs, :]
+
+            return [shuffled_feats] + inputs, self.targets
+
+        else:
+
+            features = inputs
+            stacked_feats = np.concatenate(features, axis=1)
+            shuffled_feats = stacked_feats.reshape(-1, features[0].shape[-1])
+            shuffled_idxs = np.random.permutation(shuffled_feats.shape[0])
+            shuffled_feats = shuffled_feats[shuffled_idxs, :]
+            shuffled_feats = shuffled_feats.reshape(stacked_feats.shape)
+            shuffled_feats = np.split(
+                shuffled_feats, np.cumsum([y.shape[1] for y in features])[:-1], axis=1
+            )
 
-        return [shuffled_feats] + inputs, self.targets
+            return shuffled_feats + features, self.targets[: stacked_feats.shape[0], :]
```

Comment:
If the batch size doesn't exactly divide the number of nodes then the last batch of an epoch will be smaller - `num_nodes % batch_size`. So that slicing it just there for the last epoch. It would be alot nicer though if this wasn't the case!

Reply:
I would've thought that it would already be sliced by the `GraphSAGENodeGenerator` to have a matching length, though, because the targets array should have exactly one row for each node in the batch?

Reply:
The `targets` array in this case is created in `CorruptedNodeSequence`, and is more like a batch targets array (because each node has the exact same targets).

Reply:
Oh, right!